nautobot/docs/plugins/development.md

```diff
@@ -609,6 +610,89 @@ secrets_providers = [ConstantValueSecretsProvider]
 
 After installing and enabling your plugin, you should now be able to navigate to `Secrets > Secrets` and create a new Secret, at which point `"constant-value"` should now be available as a new secrets provider to use.
 
+### Extending Filters
+
+Plugins can extend the current FilterSet's and FilterForm's that are natively provided.
+
+The basic requirements (additional requirements for each are described below) to extend a filter set or a filter form are:
+
+* The file must be named `filter_extensions.py`
+* The variable `filter_extensions` must be declared in that file, and contain a list of `PluginFilterExtension` subclasses
+* The `model` attribute of each `PluginFilterExtension` subclass must be set to a valid model name in the dotted format
+
+Nautobot dynamically creates many additional filters based upon the defined filter type. Specifically, there are additional `lookup_expr` that are created when there is neither a `lookup_expr` nor `method` parameter set on the filter already. Nautobot will also add the negation of any fields, meaning not only will `icontains` be added but so will `not icontains` using the `ic` and `nic` keys respectively.
+
+The additional lookup methods added can be found here in [nautobot/utilities/constants.py](https://github.com/nautobot/nautobot/blob/main/nautobot/utilities/constants.py), the mapping logic can be found in [nautobot/utilities/filters.py](https://github.com/nautobot/nautobot/blob/main/nautobot/utilities/filters.py).
+
+!!! tip
+    For developers of plugins that define their own model filters, note that the above are dynamically added, as long as the class correctly inherits from `nautobot.utilities.filters.BaseFilterSet`.
+
+However, that does not cover every possible use case, to list a few examples:
+
+* Usage of a custom "method", which would allow arbitrary filtering. (This is how the "q" search logic is currently performed).
```

Comment:
```suggestion
* Usage of a custom `method` argument on a filter that points to a `FilterSet` method, which would allow arbitrary filtering using custom logic. This is how the `q` field search logic is currently performed.
```